code/datums/components/simple_access.dm

```diff
@@ -0,0 +1,31 @@
+///This component allows us to give a mob access without giving them an ID card.
+/datum/component/simple_access
+	///list of accesses we are allowed to access via this component
+	var/list/access
+
+/datum/component/simple_access/Initialize(list/new_access, atom/donor_atom)
+	. = ..()
+	if(!ismob(parent))
+		return COMPONENT_INCOMPATIBLE
+	access = new_access
+	RegisterSignal(parent, COMSIG_MOB_TRIED_ACCESS, .proc/on_tried_access)
+	if(!donor_atom)
+		return
+	if(istype(donor_atom, /obj/item/organ))
+		RegisterSignal(donor_atom, COMSIG_ORGAN_REMOVED, .proc/on_donor_removed)
+	else if(istype(donor_atom, /obj/item/implant))
+		RegisterSignal(donor_atom, COMSIG_IMPLANT_REMOVED, .proc/on_donor_removed)
+
+/datum/component/simple_access/proc/on_tried_access(datum/source, atom/locked_thing)
+	SIGNAL_HANDLER
+	if(!isobj(locked_thing))
+		return LOCKED_ATOM_INCOMPATIBLE
+	var/obj/locked_object = locked_thing
+	if(locked_object.check_access_list(access))
+		return ACCESS_ALLOWED
+	else
+		return ACCESS_DISALLOWED
+
+/datum/component/simple_access/proc/on_donor_removed(datum/source)
+	SIGNAL_HANDLER
+	qdel(src)
```

Comment:
This looks wrong. It should unregister the COMSIG_MOB_TRIED_ACCESS signal that the non-existant `on_donor_added` proc registered, not delete the entire component.

Reply:
Why keep the useless component around?

Reply:
Just remove this proc and the lines concerning implants and organs from the component then. If we are going to have to instantiated the component each time the implant or organ is implanted on a mob lest he stops receiving the extra access from secondhand implants and organs, we may as well just add the component directly to the mob and store the reference of the component in a variable of the implanted item so it can be deleted later on removal. It's less confusing than having it split between files.